openmdao/error_checking/check_config.py

```diff
@@ -218,20 +291,55 @@ def _check_hanging_inputs(problem, logger):
         input_srcs = problem.model._conn_global_abs_in2out
 
     prom_ins = problem.model._var_allprocs_prom2abs_list['input']
+    abs2meta = problem.model._var_allprocs_abs2meta
     unconns = []
+    nwid = uwid = 0
+
     for prom, abslist in iteritems(prom_ins):
         unconn = [a for a in abslist if a not in input_srcs or len(input_srcs[a]) == 0]
         if unconn:
-            unconns.append(prom)
+            w = max([len(u) for u in unconn])
+            if w > nwid:
+                nwid = w
+            units = [abs2meta[a]['units'] if a in abs2meta else '' for a in unconn]
+            units = [u if u is not None else '' for u in units]
+            lens = [len(u) for u in units]
+            if lens:
+                u = max(lens)
+                if u > uwid:
+                    uwid = u
+            unconns.append((prom, unconn, units))
 
     if unconns:
+        template_abs = "{:<{nwid}} {:<{uwid}} {}\n"
+        template_prom = "{:<{nwid}} {:<{uwid}} {}\n"
         msg = ["The following inputs are not connected:\n"]
-        for prom in sorted(unconns):
-            absnames = prom_ins[prom]
-            if len(absnames) == 1 and prom == absnames[0]:  # not really promoted
-                msg.append("   %s\n" % prom)
-            else:  # promoted
-                msg.append("   %s: %s\n" % (prom, prom_ins[prom]))
+        with printoptions(precision=3):
+            for prom, absnames, units in sorted(unconns, key=lambda x: x[0]):
+                if len(absnames) == 1 and prom == absnames[0]:  # not really promoted
+                    a = absnames[0]
+                    msg.append("   " +
+                               template_abs.format(a, units[0],
+                                                   _trim_str(problem.get_val(a, get_remote=True),
+                                                              25),
+                                                   nwid=nwid + 3, uwid=uwid))
+                else:  # promoted
+                    vals = [problem.get_val(a, get_remote=True) for a in absnames]
+                    mismatch = _has_val_mismatch(problem.model._var_allprocs_discrete['input'],
+                                                 absnames, units, vals)
+                    if mismatch:
+                        msg.append("\n   ----- WARNING: inconsistent units and/or values!! -----\n")
+                    msg.append("   {}  (p):\n".format(prom))
+                    for a, u, v in zip(absnames, units, vals):
+                        msg.append("      " +
+                                   template_prom.format(a, u,
+                                                        _trim_str(problem.get_val(a,
+                                                                                   get_remote=True),
+                                                                   25),
```

Comment:
This is really hard to read. Could you create a temporary variable to hold `_trim_str(problem.get_val(a,get_remote=True)` to improve readability? Added advantage, looks like you can do it outside of the if branch.

Reply:
fixed